Std/Data/Nat/Lemmas.lean

```diff
@@ -991,3 +1002,36 @@ theorem shiftRight_eq_div_pow (m : Nat) : ∀ n, m >>> n = m / 2 ^ n
   | k + 1 => by
     rw [shiftRight_add, shiftRight_eq_div_pow m k]
     simp [Nat.div_div_eq_div_mul, ← Nat.pow_succ]
+
+/-! ## deprecated -/
+
+@[deprecated Nat.le_sub_iff_add_le]
+protected theorem add_le_to_le_sub (n : Nat) (h : m ≤ k) : n + m ≤ k ↔ n ≤ k - m :=
+  (Nat.le_sub_iff_add_le h).symm
+
+@[deprecated Nat.add_one_sub_one]
+theorem succ_sub_one (n) : succ n - 1 = n := rfl
+
+@[deprecated Nat.le_of_sub_le_sub_left]
+protected theorem le_of_le_of_sub_le_sub_left {m : Nat} : n ≤ k → k - m ≤ k - n → n ≤ m :=
+  Nat.le_of_sub_le_sub_left
+
+@[deprecated Nat.le_of_sub_le_sub_right]
+protected theorem le_of_le_of_sub_le_sub_right {k : Nat} : k ≤ m → n - k ≤ m - k → n ≤ m :=
+  Nat.le_of_sub_le_sub_right
+
+@[deprecated Nat.add_le_of_le_sub']
+protected theorem add_le_of_le_sub_left {n k m : Nat} (h : m ≤ k) : n ≤ k - m → m + n ≤ k :=
+  Nat.add_le_of_le_sub' h
+
+@[deprecated Nat.sub_lt_of_pos_le]
+protected theorem sub_lt_self (h₀ : 0 < a) (h₁ : a ≤ b) : b - a < b :=
+  Nat.sub_lt_of_pos_le h₀ h₁
+
+@[deprecated]
+theorem sub_lt_succ (a b) : a - b < succ a :=
+  lt_succ_of_le (sub_le a b)
+
+@[deprecated]
+theorem sub_one_sub_lt (h : i < n) : n - 1 - i < n := by
```

Comment:
Why is this deprecated?

Reply:
Same.

Reply:
This one looks relevant for indexing into arrays

Reply:
Agreed. I suspect it has a better name somewhere else but I never tried looking for it.